sdk/keyvault/azure-keyvault-keys/migration_guide.md

```diff
@@ -0,0 +1,271 @@
+crypto_client = CryptographyClient(key=key, credential=credential)
+```
+
+### Async operations
+
+The modern `azure-keyvault-keys` library includes a complete async API supported on Python 3.5+. To use it, you must first install an async transport, such as [aiohttp](https://pypi.org/project/aiohttp/). See [azure-core documentation](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md#transport) for more information.
+
+Async operations are available on async clients, which should be closed when they're no longer needed. Each async client is an async context manager and defines an async `close` method. For example:
+
+```python
+from azure.identity.aio import DefaultAzureCredential
+from azure.keyvault.keys.aio import KeyClient
+
+credential = DefaultAzureCredential()
+
+# call close when the client is no longer needed
+client = KeyClient(vault_url="https://my-key-vault.vault.azure.net/", credential=credential)
+...
+await client.close()
+
+# alternatively, use the client as an async context manager
+client = KeyClient(vault_url="https://my-key-vault.vault.azure.net/", credential=credential)
+async with client:
+  ...
+```
+
+### Create a key
+
+In `azure-keyvault` you could create a key by using `KeyVaultClient`'s `create_key` method, which required a vault endpoint, key name, and key type. This method returned a `KeyBundle` containing the key.
+
+```python
+# create an RSA key
+key_bundle = client.create_key(
+    vault_base_url="https://my-key-vault.vault.azure.net/",
+    key_name="key-name",
+    kty="RSA"
+)
+key = key_bundle.key
+
+# create an elliptic curve key
+key_bundle = client.create_key(
+    vault_base_url="https://my-key-vault.vault.azure.net/",
+    key_name="key-name",
+    kty="EC"
+)
+key = key_bundle.key
+```
+
+Now in `azure-keyvault-keys` there are multiple ways to create keys. You can provide a key name and type to the general `create_key` method, or provide just a name to `create_rsa_key` or `create_ec_key`. These methods all return the created key as a `KeyVaultKey`.
+
+```python
+# create a key with specified type
+key = key_client.create_key(name="key-name", key_type="oct")
+print(key.name)
+print(key.key_type)
+
+# create an RSA key
+rsa_key = key_client.create_rsa_key(name="rsa-key-name", size=2048)
+
+# create an elliptic curve key
+ec_key = key_client.create_ec_key(name="ec-key-name", curve="P-256")
+```
+
+### Retrieve a key
+
+In `azure-keyvault` you could retrieve a key (in a `KeyBundle`) by using `get_key` and specifying the desired vault endpoint, key name, and key version. You could retrieve the versions of a key with the `get_key_versions` method, which returned an iterator-like object.
+
+```python
+from azure.keyvault import KeyId
+
+key_items = client.get_key_versions(
+    vault_base_url="https://my-key-vault.vault.azure.net/",
+    key_name="key-name"
+)
+
+for key_item in key_items:
+    key_id = KeyId(key_item.kid)
```

Comment:
This isn't mentioned in the text or used in the example?

Reply:
This variable isn't used directly, but I split up the process of fetching the key version because I wanted to clearly show how the `KeyId` was being used. I don't think condensing lines 159/160 into `key_version = KeyId(key_item.kid).version` would be too complicated though, if you think that's neater

Reply:
Ah, my bad, it's used in the sense I meant on the next line, it's fine like this 🤓